File: integration-prototypes/csharp/kafka-single/tests/KafkaSingleClients.Tests/KafkaSingleTests.cs
using KafkaSingleClients;

namespace KafkaSingleClients.Tests;

public sealed class KafkaSingleTests
{
    [Fact]
    public void BuildMessage_FormatsExpectedJson()
    {
        Assert.Equal("{\"id\":\"1\",\"text\":\"hello\"}", ProducerClient.BuildMessage("1", "hello"));
    }
}

File: integration-prototypes/java/kafka-single/src/test/java/org/prototype/kafka/single/KafkaSingleTests.java
package org.prototype.kafka.single;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public final class KafkaSingleTests {
    @Test
    void buildMessageFormatsJson() {
        assertEquals("{\"id\":\"1\",\"text\":\"hello\"}", ProducerClient.buildMessage("1", "hello"));
    }
}
